src/radio/radio.js

```diff
@@ -5,5 +5,161 @@ This source code is licensed under the MIT license found in the
+    this.setState({isHovered: true});
+    this.props.onMouseEnter && this.props.onMouseEnter(e);
+  };
+
+  onMouseLeave = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({isHovered: false, isActive: false});
+    this.props.onMouseLeave && this.props.onMouseLeave(e);
+  };
+
+  onMouseDown = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({isActive: true});
+    this.props.onMouseDown && this.props.onMouseDown(e);
+  };
+
+  onMouseUp = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({isActive: false});
+    this.props.onMouseUp && this.props.onMouseUp(e);
+  };
+
+  onFocus = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({isFocused: true});
+    this.props.onFocus && this.props.onFocus(e);
+  };
+
+  onBlur = (e: SyntheticInputEvent<HTMLInputElement>) => {
+    this.setState({isFocused: false});
+    this.props.onBlur && this.props.onBlur(e);
+  };
+
+  render() {
+    const {overrides = {}} = this.props;
+    const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
+    const [Label, labelProps] = getOverrides(overrides.Label, StyledLabel);
+    const [Input, inputProps] = getOverrides(overrides.Input, StyledInput);
+    const [RadioMarkInner, radioMarkInnerProps] = getOverrides(
+      overrides.RadioMarkInner,
+      StyledRadioMarkInner,
+    );
+    const [RadioMarkOuter, radioMarkOuterProps] = getOverrides(
+      overrides.RadioMarkOuter,
+      StyledRadioMarkOuter,
+    );
+
+    if (__DEV__) {
+      if (this.props.overrides && this.props.overrides.RadioMark) {
+        // eslint-disable-next-line no-console
+        console.warn(`The RadioMark prop will be deprecated in the next major version. Please use
+          the 'RadioMarkInner', and 'RadioMarkOuter' overrides.
+        `);
+      }
+    }
+
+    const sharedProps = {
+      $checked: this.props.checked,
+      $disabled: this.props.disabled,
+      $isActive: this.state.isActive,
+      $isError: this.props.isError,
+      $isFocused: this.state.isFocused,
+      $isHovered: this.state.isHovered,
+      $labelPlacement: this.props.labelPlacement,
+      $required: this.props.required,
+      $value: this.props.value,
+    };
+
+    const label = (
+      <Label {...sharedProps} {...labelProps}>
+        {this.props.children}
+      </Label>
+    );
+
+    return (
+      <Root
+        onMouseEnter={this.onMouseEnter}
+        onMouseLeave={this.onMouseLeave}
+        onMouseDown={this.onMouseDown}
+        onMouseUp={this.onMouseUp}
+        {...sharedProps}
+        {...rootProps}
+      >
+        {isLabelTopLeft(this.props.labelPlacement) && label}
+        <RadioMarkOuter {...sharedProps} {...radioMarkOuterProps}>
+          <RadioMarkInner {...sharedProps} {...radioMarkInnerProps} />
+        </RadioMarkOuter>
+        <Input
+          aria-invalid={this.props.isError || null}
+          aria-required={this.props.isError || null}
```

Comment:
Do we want to introduce `required` prop for it instead of setting required based on error state?